config/config.json

```diff
@@ -16,220 +16,27 @@
-        "misc-move-forwarding-reference" : ["default", "sensitive", "extreme"],
-        "misc-multiple-statement-macro" : ["sensitive", "extreme"],
-        "performance-noexcept-move-constructor" : ["default", "sensitive", "extreme"],
-        "misc-redundant-expression" : ["default", "sensitive", "extreme"],
-        "misc-sizeof-container" : ["default", "sensitive", "extreme"],
-        "misc-sizeof-expression" : ["default", "sensitive", "extreme"],
-        "misc-string-compare" : ["default", "sensitive", "extreme"],
-        "misc-string-constructor" : ["default", "sensitive", "extreme"],
-        "misc-string-integer-assignment" : ["sensitive", "extreme"],
-        "misc-string-literal-with-embedded-nul" : ["default", "sensitive", "extreme"],
-        "misc-suspicious-enum-usage" : ["default", "sensitive", "extreme"],
-        "misc-suspicious-missing-comma" : ["default", "sensitive", "extreme"],
-        "misc-suspicious-semicolon" : ["default", "sensitive", "extreme"],
-        "misc-suspicious-string-compare" : ["sensitive", "extreme"],
-        "misc-swapped-arguments" : ["default", "sensitive", "extreme"],
-        "misc-unconventional-assign-operator" : ["default", "sensitive", "extreme"],
-        "misc-uniqueptr-reset-release" : ["default", "sensitive", "extreme"],
-        "misc-unused-alias-decls" : ["sensitive", "extreme"],
-        "misc-unused-parameters" : ["extreme"], 
-        "misc-unused-raii" : ["default", "sensitive", "extreme"],
-        "misc-unused-using-decls" : ["sensitive", "extreme"],
-        "misc-use-after-move" : ["default", "sensitive", "extreme"],
-        "misc-virtual-near-miss" : ["default", "sensitive", "extreme"],
-        "modernize-deprecated-headers" : ["sensitive", "extreme"],
-        "modernize-replace-auto-ptr" : ["sensitive", "extreme"],
-        "modernize-replace-random-shuffle" : ["sensitive", "extreme"],
-        "mpi-buffer-deref" : ["sensitive", "extreme"],
-        "mpi-type-mismatch" : ["sensitive", "extreme"],
-        "performance-faster-string-find" : ["sensitive", "extreme"],
-        "performance-for-range-copy" : ["sensitive", "extreme"],
-        "performance-implicit-conversion-in-loop" : ["sensitive", "extreme"],
-        "performance-inefficient-algorithm" : ["default", "sensitive", "extreme"],
-        "performance-inefficient-string-concatenation" : ["sensitive", "extreme"],
-        "performance-inefficient-vector-operation" : ["sensitive", "extreme"],
-        "performance-move-const-arg" : ["default", "sensitive", "extreme"],
-        "performance-noexcept-move-constructor" : ["default", "sensitive", "extreme"],
-        "performance-type-promotion-in-math-fn" : ["sensitive", "extreme"],
-        "performance-unnecessary-copy-initialization" : ["sensitive", "extreme"],
-        "performance-unnecessary-value-param" : ["sensitive", "extreme"]
-      }
-    }
+      "clangsa" : [{
+          "config_name": "default",
```

Comment:
Why do we need this? Can not we use the version regex to identify a checker config?

Reply:
I think a config name and a detection regex is descriptive in this case. config_name is just a label, not used anywhere, maybe only in logging.

Reply:
I still think a config name is a good option if the regex happens to be too arcane to mention in human-readable context.